### PR TITLE
Improve source-mapping

### LINKED ISSUE #798: Improve source mapping in Kontrol
Check with @lisandrasilva regarding Simbolik approach to handling source maps and port it to Kontrol.

### PR DESCRIPTION
**Pull request by lisandrasilva**
This PR imports Simbolik approach to source map bytecode into solidity code. 
Fixes #798.

### CODE REVIEW

File: src/kontrol/solc.py [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+import os
+from collections import deque
+from functools import cache, cached_property
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from pathlib import Path
+
+import pyevmasm  # type: ignore
```
Comment (anvacaru) [inline src/kontrol/solc.py:15]: Can we remove this?

```suggestion
import pyevmasm
```

File: src/kontrol/foundry.py [hunk 2]
```diff
@@ -362,16 +366,26 @@ def short_info_for_contract(self, contract_name: str, cterm: CTerm) -> list[str]
...
+                if type(program_cell) is KToken:
+                    try:
+                        bytecode = ast.literal_eval(program_cell.token)
+                        instruction = compilation_unit.get_instruction(bytecode, int(pc_cell.token))
+                        node = instruction.node()
+                        start_line, _, end_line, _ = node.source_range()
+                        return self.solidity_src_print(Path(node.source.name), start_line - 1, end_line)
+                    except Exception as e:
+                        return [str(e)]
+                return ['NO DATA 2']
```
Comment (palinatolmach) [inline src/kontrol/foundry.py:383]: Same question about the typo:
```suggestion
                return ['NO DATA 2']
```

**Review by palinatolmach — COMMENTED** [review]
Can you please add an example of how the result of this change looks like to the issue description? E.g., an updated section in the KCFG view.

File: src/kontrol/foundry.py [hunk 3]
```diff
@@ -362,16 +366,26 @@ def short_info_for_contract(self, contract_name: str, cterm: CTerm) -> list[str]
...
         elif type(element) is KCFG.Edge:
             return list(element.rules)
         elif type(element) is KCFG.NDBranch:
             return list(element.rules)
-        return ['NO DATA']
+        return ['NO DATA 3']
```
[the remark this answers is not in the document]
Comment (lisandrasilva) [reply src/kontrol/foundry.py:388]: Yes, it is. I will remove it. Thank you!

File: src/kontrol/solc.py (hunk 1 above)
Comment (lisandrasilva) [reply src/kontrol/solc.py:15]: No, otherwise we get an error.

File: src/kontrol/foundry.py (hunk 2 above)
Comment (lisandrasilva) [reply src/kontrol/foundry.py:383]: Do you mean to add in issue #798 an example of how the source mapping is working after these changes?

**Comment by palinatolmach** [comment]
Closed as subsumed by https://github.com/runtimeverification/kontrol/pull/820.

**Pull request closed without merging**